be2-scala/src/test/scala/ch/epfl/pop/storage/DbActorNewSuite.scala

```diff
@@ -1,9 +1,277 @@
+    storage.elements(channelName2) should equal (ChannelData(ObjectType.LAO, message2.message_id :: Nil).toJsonString)
+    storage.elements(s"$channelName2:${message2.message_id}") should equal (message2.toJsonString)
   }
+
+  test("createChannel effectively creates a new channel") {
+    val storage: InMemoryStorage = InMemoryStorage()
+    val dbActor: ActorRef = system.actorOf(Props(DbActorNew(mediatorRef, storage)), "r")
+
+    storage.size should equal (0)
+
+    dbActor ! DbActorNew.CreateChannel(Channel(CHANNEL_NAME), ObjectType.LAO); sleep()
+
+    expectMsg(DbActorNew.DbActorAck())
+    storage.size should equal (1)
+    storage.elements(CHANNEL_NAME) should equal (ChannelData(ObjectType.LAO, Nil).toJsonString)
+  }
+
+  test("createChannel does not overwrite channels on duplicates") {
+    val storage: InMemoryStorage = InMemoryStorage()
+    val dbActor: ActorRef = system.actorOf(Props(DbActorNew(mediatorRef, storage)))
+
+    storage.size should equal (0)
+
+    dbActor ! DbActorNew.CreateChannel(Channel(CHANNEL_NAME), ObjectType.LAO); sleep()
+
+    expectMsg(DbActorNew.DbActorAck())
+    storage.size should equal (1)
+    storage.elements(CHANNEL_NAME) should equal (ChannelData(ObjectType.LAO, Nil).toJsonString)
+
+    dbActor ! DbActorNew.CreateChannel(Channel(CHANNEL_NAME), ObjectType.LAO); sleep()
+
+    // expectMsg(DbActorNew.DbActorNAck())
+    storage.size should equal (1)
+    storage.elements(CHANNEL_NAME) should equal (ChannelData(ObjectType.LAO, Nil).toJsonString)
+  }
+
+  test("createChannelsFromList creates multiple channels") {
+    val storage: InMemoryStorage = InMemoryStorage()
+    val dbActor: ActorRef = system.actorOf(Props(DbActorNew(mediatorRef, storage)))
+
+    storage.size should equal (0)
+
+    dbActor ! DbActorNew.CreateChannelsFromList((Channel(CHANNEL_NAME), ObjectType.ELECTION) :: (Channel(s"${CHANNEL_NAME}2"), ObjectType.ROLL_CALL) :: Nil); sleep()
+
+    expectMsg(DbActorNew.DbActorAck())
+    storage.size should equal (2)
+    storage.elements(CHANNEL_NAME) should equal (ChannelData(ObjectType.ELECTION, Nil).toJsonString)
+    storage.elements(s"${CHANNEL_NAME}2") should equal (ChannelData(ObjectType.ROLL_CALL, Nil).toJsonString)
+  }
+
+  test("createChannelsFromList does not overwrite channels on duplicates (duplicates already created)") {
+    val storage: InMemoryStorage = InMemoryStorage()
+    val dbActor: ActorRef = system.actorOf(Props(DbActorNew(mediatorRef, storage)))
+
+    storage.size should equal (0)
+
+    dbActor ! DbActorNew.CreateChannel(Channel(CHANNEL_NAME), ObjectType.LAO); sleep()
+
+    expectMsg(DbActorNew.DbActorAck())
+    storage.size should equal (1)
+
+    // adding 2 new channels (one of which already exists)
+    dbActor ! DbActorNew.CreateChannelsFromList((Channel(CHANNEL_NAME), ObjectType.ELECTION) :: (Channel(s"${CHANNEL_NAME}2"), ObjectType.ROLL_CALL) :: Nil); sleep()
```

Comment:
this seems precisely a case that should fail.
if `CHANNEL_NAME` was already added as `ObjectType.LAO`, how can we allow it to be created as `ObjectType.ELECTION` without failure ?